Delete deprecated HWReport

saxocellphone opened the pull request:
Closes #1987. HWReport is only used in ElectornicGraderController and ReportController. I tested both of them out and nothing seems to break.

CouleeApps commented:
How does this interact with `Gradeable::$grade_file`? If you're deleting the code that generates those files you should clean up the code that tries to load them.

saxocellphone commented:
@MasterOdin  In the file $grade_file is also used for queueing submissions. I talked to Barb about it and she said those should be named grading_queue_files, and I deleted the portion related to HWReports. I also replaced hasGradeFile with beenTAGraded since they are the same in functionality.
